osquery/tables/applications/darwin/browser_plugins.cpp

```diff
@@ -227,26 +281,204 @@ QueryData genSafariExtensions(QueryContext& context) {
       continue;
     }
 
-    auto dir = fs::path(directory->second) / kSafariExtensionsPath;
+    auto dir = fs::path(directory->second) / kSafariLegacyExtensionsPath;
     // Check that an extensions directory exists.
     if (!pathExists(dir).ok()) {
       continue;
     }
 
     // Glob the extension files.
     std::vector<std::string> paths;
-    if (!resolveFilePattern(
-             dir / kSafariExtensionsPattern, paths, GLOB_ALL | GLOB_NO_CANON)
+    if (!resolveFilePattern(dir / kSafariLegacyExtensionsPattern,
+                            paths,
+                            GLOB_ALL | GLOB_NO_CANON)
              .ok()) {
       continue;
     }
 
     for (const auto& extension_path : paths) {
-      genSafariExtension(uid->second, gid->second, extension_path, results);
+      genSafariLegacyExtension(
+          uid->second, gid->second, extension_path, results);
     }
   }
+}
 
-  return results;
+inline bool getPtreeFromPlist(const fs::path& plist_path, pt::ptree& tree) {
+  // Perform a dry run of the file read.
+  if (!isReadable(plist_path).ok()) {
+    return false;
+  }
+
+  // Read the plist file
+  std::string content;
+  if (!readFile(plist_path, content).ok()) {
+    return false;
+  }
+
+  // Parse the plist content
+  if (!parsePlistContent(content, tree).ok()) {
+    return false;
+  }
+
+  return true;
+}
+
+inline void getSandboxedExtensionData(const std::string& extension_path,
+                                      const std::string& plist_path,
+                                      SandboxedExtensionsData& data) {
+  // Parse the plist content
+  pt::ptree tree;
+  if (!getPtreeFromPlist(plist_path, tree)) {
+    return;
+  }
+
+  // Populating SandboxedExtensionData entry
+  SandboxedExtensionData entry;
+  entry.app_path = extension_path;
+  entry.identifier = tree.get<std::string>("CFBundleIdentifier", "");
+  entry.name = tree.get<std::string>("CFBundleDisplayName", "");
+  entry.sdk = tree.get<std::string>("CFBundleInfoDictionaryVersion", "");
+  entry.version = tree.get<std::string>("CFBundleShortVersionString", "");
+  data.push_back(entry);
```

Comment:
For what it's worth, we also pull these fields from the plist:
* `NSHumanReadableDescription`
* `CFBundleVersion`
* `NSHumanReadableCopyright`

I don't know if it makes sense to grab them, or let people join on `plist` if they want extra info

Reply:
Thanks, I've just included these fields on the list parsing